### DISCUSSION TITLE
How to know when an `AppNotification` is dismissed?

### DISCUSSION THREAD

**Discussion by stevenbrix** [microsoft/WindowsAppSDK#4224] [Q&A]
The actual `Windows.UI.Toast.ToastNotification` object has events to know when the toast has been dismissed, but there isn't anything like this on the `AppNotification` class. Is there a way to know when the user has dismissed the notification?

**Comment by stevenbrix** [comment]
@DrusTheAxe do you have any ideas here?

**Comment by stevenbrix** [comment]
not a great answer, but one thing that can be done in a packaged application is to use the `ToastNotification` class:

```
var appNotification = new AppNotificationBuilder()
    .AddArgument("action", "ToastClick")
    .AddArgument(Common.scenarioTag, ScenarioId.ToString())
    .SetAppLogoOverride(new System.Uri("file://" + App.GetFullPathToAsset("Square150x150Logo.png")), AppNotificationImageCrop.Circle)
    .AddText(ScenarioName)
    .AddText("This is an example message using XML")
    .AddButton(new AppNotificationButton("Open App")
        .AddArgument("action", "OpenApp")
        .AddArgument(Common.scenarioTag, ScenarioId.ToString()))
    .BuildNotification();

    var notifier = ToastNotificationManager.CreateToastNotifier(AppInfo.Current.AppUserModelId);
    var doc = new XmlDocument();
    doc.LoadXml(appNotification.Payload);
    var notification = new ToastNotification(doc);
    notification.Dismissed += Notification_Dismissed;
    notifier.Show(notification);
 ```
 
 i wonder if this would work if packaging with external location as well

**Reply by danielayala94** [reply]
Packaged apps don't need to call CreateToastNotifier(string). CreateToastNotifier() with no parameters will suffice.

**Comment by danielayala94** [comment]
As correctly pointed out in the question, Windows App SDK App Notifications doesn't support events.

Currently, the only possible workaround is to use the Windows SDK Toast APIs -> Dismissed event. The code provided by @stevenbrix is mostly the way to go, plus the API clarification I made in his comment.
